x/supplier/keeper/msg_server_unstake_supplier.go

```diff
@@ -36,6 +36,18 @@ func (k msgServer) UnstakeSupplier(
 		logger.Info(fmt.Sprintf("Supplier not found. Cannot unstake address %s", msg.Address))
 		return nil, types.ErrSupplierNotFound
 	}
+
```

Comment:
_tl;dr Operator address can unstake too._

----

I know I reference this in the past, but going to call this out VERY explicitly.

If you run (on Morse), `pocket nodes stake help non-custodial` for the short version, this what is says:

```
non-custodial Stake a node in the network, non-custodial stake allows a different output address for rewards/return of staked funds. The signer may be the operator or the output address. The signer must specify the public key of the operator`
```

If you run `pocket nodes stake non-custodial --help` for the long version, this is what it says:

```
Will prompt the user for the signer account passphrase, fund and fees are
collected from signer account.  If the node is already staked, this transaction
acts as an *update* transaction.  A node can update relayChainIDs, serviceURI,
and raise the stake amount with this transaction.  The output address can be
changed when the output address is empty or a transaction is signed by the
current output address.
```

As an analogy, imagine you are paying someone to drive you. Both you and the driver can say "STOP", but if the ride ends early (regardless of who says stop), you get your money back.


I highly recommend you run all of these commands and deeply understand the help commands.

```
pocket nodes stake non-custodial --help
pocket nodes stake help non-custodial

pocket nodes stake custodial --help
pocket nodes stake help custodial

pocket nodes stakeNew --help
pocket nodes help stakeNew
```

Reply:
Now, the behavior is the same as in Morse.